src/main/java/net/openhft/chronicle/core/pool/StringBuilderPool.java

```diff
@@ -39,10 +59,23 @@ public class StringBuilderPool {
      * The length of the StringBuilder is reset to 0 before being returned.
      *
      * @return a StringBuilder instance with length 0.
+     * @deprecated Use {@link #acquireStringBuilderScoped()} instead
      */
+    @Deprecated(/* To be removed in x.26 */)
     public StringBuilder acquireStringBuilder() {
         StringBuilder sb = sbtl.get();
         sb.setLength(0);
         return sb;
     }
+
+    /**
+     * Returns a scoped, thread-local StringBuilder instance from the pool.
+     * If the current thread does not yet have a StringBuilder, it is created and added to the pool.
+     * The length of the StringBuilder is reset to 0 before being returned.
+     *
+     * @return a StringBuilder instance with length 0.
+     */
+    public ScopedResource<StringBuilder> acquireStringBuilderScoped() {
+        return scopedStringBuilderPool.get();
+    }
```

Comment:
We could potentially deprecate the whole StringBuilderPool and just use a ScopedThreadLocal instead

Reply:
works for me

Reply:
Done, `StringBuilderPool` will just provide a factory method going forward `StringBuilderPool#createThreadLocal`